### DISCUSSION TITLE
The skywalking java agent cannot collect spring sleuth data

### DISCUSSION THREAD

**Discussion by beat1030** [apache/skywalking#12773] [General]
My SpringCloud project added the agent configuration in accordance with the official guidelines when java was started, but I found a problem that java agent could collect other indicators of the service, but could not collect spring sleuth indicators, spring in SkyWalking Ui sleuth data is empty, how can I configure to see some http metrics
![1](https://github.com/user-attachments/assets/4bf17d2a-0488-48a7-990d-79f16d5b8e4d)
imprint
skywalking v10.1.0
java agent v9.3.0
spring boot 2.3.12.RELEASE
spring cloud Hoxton.SR12

**Comment by wu-sheng** [comment]
I can't see your screenshot. Slueth is an API lib, you need to connect them with us, such as Micrometer. 
I don't know what do you want.
But maybe these docs help
- https://skywalking.apache.org/docs/skywalking-java/latest/en/setup/service-agent/java-agent/application-toolkit-micrometer/
- https://skywalking.apache.org/docs/skywalking-java/latest/en/setup/service-agent/java-agent/application-toolkit-micrometer-1.10/

**Reply by beat1030** [reply]
Do you mean that if you want to see service->instance->spring sleuth in skywalking UI to see relevant indicator data, do springcloud projects need to add the following pom dependencies? What else do I need to do?

![17315694614592](https://github.com/user-attachments/assets/53375276-9aad-48b5-9609-903577585b3b)

**Reply by wu-sheng** [reply]
Follow the docs. I have provided you the link.

**Reply by beat1030** [reply]
Based on the information within the link you provided, I have the impression that it pertains to the configuration of custom metrics rather than the spring sleuth metric data or HTTP-related metric data of the Spring Cloud services as described in my requirements, which I expect to observe in the SkywalkingUI interface. Nevertheless, I attempted as per your guidance by introducing the 'apm-toolkit-micrometer-registry' and 'apm-toolkit-micrometer-1.10' dependencies into the Spring Cloud project, without adding any additional configurations. After the startup, the corresponding spring sleuth metric data of the service still couldn't be seen in the SkywalkingUI interface. I am unsure whether it is a configuration problem or if I failed to delineate my requirements precisely. Please elaborate or provide a demonstration case. Thank you.

**Reply by wu-sheng** [reply]
Do you follow the document to enable backend analysis?
https://skywalking.apache.org/docs/main/latest/en/setup/backend/micrometer-observations/#set-up-backend-receiver

Notice, we have default values in the YAML. You should append rather than override, which could make other metrics disappear.

**Reply by beat1030** [reply]
Hi, Mr. Wu
Based on the information you provided, I'm still not seeing the Spring Sleuth data. Let me show you the steps I followed to see if I did everything correctly.

Step 1: I added the following configurations to the application.yml file in SkyWalking.
![5](https://github.com/user-attachments/assets/7aee3130-dbb4-4752-9b22-6fa1b7865336)

Step 2: I added the necessary dependencies to the pom.xml in the Spring Cloud project and updated the startup configuration.
![2](https://github.com/user-attachments/assets/62b63e1e-b9b5-46b6-a576-f1be32bf239b)
![3](https://github.com/user-attachments/assets/e8da2788-7cbe-42d9-b73d-e27b94e5c9a1)


However, even after making these changes, I still don’t see the Spring Sleuth data in the SkyWalking UI under the Spring Sleuth item. Could you take another look to help me find the issue?
![1](https://github.com/user-attachments/assets/13c170f2-11fd-4a0d-882c-0ce44debe340)


 Thanks!

**Reply by wu-sheng** [reply]
I don't use Slueth, so, can't tell that.
But demo service was howing that, the metrics should be there.
